### PR TITLE
Add compareTo aliases

### PR DESCRIPTION
**Pull request by d-mon-**
I created the PR for now until we figure out the new name for `equals`.

It might conflict with https://github.com/srknzl/bigdecimal.js/pull/82

fixes: [#73](https://github.com/srknzl/bigdecimal.js/issues/73)

### CODE REVIEW

File: test/compareTo.js [hunk 1]
```diff
@@ -14,14 +14,6 @@ describe('Compare test', function () {
                     Big(test.args[1])
                 ).toString();
             };
-            if (test.result === 'errorThrown') {
```
Comment (srknzl) [inline test/compareTo.js:17]: I know compareToTestCases.json and some other json files does not include any "errorThrown" test cases, but can we keep this part just for consistency with other test codes? I want to be sure that if Java throws, our lib should also throw regardless of the method that is tested.

File: src/bigdecimal.ts [hunk 2]
```diff
@@ -2939,6 +2950,112 @@ export class BigDecimal {
...
+     * Alias for `compareTo(val) > 0`.
+     *
+     * @param val value to which this `BigDecimal` is to be compared.
+     * This value will be converted to a `BigDecimal` before the operation.
+     * See the {@link Big | constructor} to learn more about the conversion.
+     * @returns true if the value is greater than `val`
+     * @see     {@link compareTo}
+     * @alias   {@link gt}
+     */
+    greaterThan(val: any) {
```
Comment (srknzl) [inline src/bigdecimal.ts:2985]: Can you add return types to these methods?

File: test/equalsStrict.js [hunk 3]
```diff
@@ -4,24 +4,24 @@ const chai = require('chai');
 const testCases = require('../util/output/equalsTestCases.json');
 chai.should();
 
-describe('Equals test', function () {
+describe('Equals strict test', function () {
 
     it('should calculate equals correctly', function () {
```
Comment (srknzl) [inline test/equalsStrict.js:9]: can you rename remaining "equals" texts to "equalsStrict" in this file? 


1. in it()
2. assertion message strings

**Review by srknzl — CHANGES_REQUESTED** [review]
Looks good overall except some minor things

File: src/bigdecimal.ts [hunk 4]
```diff
@@ -2481,6 +2481,12 @@ export class BigDecimal {
      * One example that shows how 2.0 and 2.00 are **not**
      * substitutable for each other under some arithmetic operations
      * are the two expressions:
+     * new BigDecimal("2.0" ).divide(BigDecimal.valueOf(3), HALF_UP) which evaluates to 0.7 and
+     * new BigDecimal("2.00").divide(BigDecimal.valueOf(3), HALF_UP) which evaluates to 0.67.
```
Comment (srknzl) [inline src/bigdecimal.ts:2485]: This does not look good in the API documentation: 
image


We can put the code in backticks and add one or two line breaks

**Comment by srknzl** [comment]
To think more, let's don't rush to merge this PR. Maybe we can come up with something better. Current implementation is ok though.

**Comment by srknzl** [comment]
Maybe we can isolate "equals" part of this PR and merge the rest quickly.

**Comment by d-mon-** [comment]
Included in https://github.com/srknzl/bigdecimal.js/pull/99

**Pull request closed without merging**